src/artifacts_poetry_plugin/plugin.py

```diff
@@ -4,23 +4,141 @@
+    return wheel_files
+
+
+def add_package_paths(packages: List[Package], wheel_files):
+    """Adds the wheel file locations for each package's files"""
+    non_deployable_packages = []
+    deployable_packages = []
+    for package in packages:
+        file_found = False
+        for file in package.files:
+            if file["file"] in wheel_files:
+                file_found = True
+                file["url"] = wheel_files[file["file"]]
+        if file_found:
+            deployable_packages.append(package)
+        else:
+            non_deployable_packages.append(package)
+    return deployable_packages, non_deployable_packages
+
+
+def upload_packages(poetry, packages: List[Package], url, username, password, io):
+    uploader = Uploader(poetry=poetry, io=io)
+    uploader.auth(username=username, password=password)
+    session = uploader.make_session()
+    for package in packages:
+        uploader._package = package
+        for file in package.files:
+            if "url" in file:
+                uploader._upload_file(
+                    session,
+                    url,
+                    file=file["url"],
+                )
+
+
+def add_project_package(poetry, packages: List[Package]):
+    """Adds the generated package to the list of packages with its associated wheel file"""
+    project_package = poetry.package
+    for root, dirs, files in os.walk("dist/"):
+        for file in files:
+            if file.endswith(".whl"):
+                file_data = {"file": file, "url": Path(os.path.join(root, file))}
+                project_package.files.append(file_data)
+    packages.append(project_package)
+
+
+def get_deploy_packages(poetry, cache_dir):
+    """Gets all deployable packages from the given cache directory. A package is deployable if it has any wheel files associated with it"""
+    packages = get_packages(poetry)
+    wheel_files = get_cached_wheel_files(cache_dir)
+    deployable_packages, non_deployable_packages = add_package_paths(
+        packages=packages, wheel_files=wheel_files
+    )
+    add_project_package(poetry, deployable_packages)
+    return deployable_packages, non_deployable_packages
+
+
 class ArtifactsDeploy(Command):
-    name = ARTIFACTS_LIST
+    name = ARTIFACTS_DEPLOY
+
+    arguments = [
+        argument(REPOSITORY_NAME, "The name of the repository to deploy artifacts")
+    ]
+
+    help = "The Artifacts Deploy command is used to deploy all of a Python project's dependencies to a repository"
 
     def handle(self) -> int:
-        self.line("Artifacts List")
-        packages = getPackages(Factory().create_poetry(Path(os.getcwd())))
-        print(packages)
+        self.line("Artifacts Deploy")
+        repo_name = self.argument(REPOSITORY_NAME)
+        poetry = Factory().create_poetry(Path(os.getcwd()))
+        config = poetry.config
+        repo_data = get_repository_data(config, repo_name, self.io)
+        url = repo_data["url"]
```

Comment:
A: Might be worth wrapping this in a class so we aren't depending on strings to be the same in two different areas of code